Real content

otaviobp opened the pull request:
I've been running those tests for currently release and they are quite stable right now. Sending for review

jwakre commented on test/real_content/swupd_update.bats:
```diff
@@ -1,156 +1,13 @@
-
-@test "test bundle-list --all" {
+# Test if bundle-list --all is working because all other tests depend on that
+@test "RC001: test bundle-list --all" {
```
This file is called swupd_update, but this is a bundle-list test. Does it make sense to move this one?

otaviobp replied:
The only reason I'm testing a bundle-list here is that if bundle-list is failing we can't trust the results of the tests. And I'd like to run that before next test.

castulo replied:
It is not a good practice to have tests that depend from other tests, but sometimes it can be necessary. So if you think the result of tests RC002-04 depend on the good execution of RC001 maybe you can create a file like .RC001 at the end of the first test, and run the following tests only if .RC001 exists. Then you can remove .RC001 in the global_teardown (and in the global_setup in case it was left over for some reason).

otaviobp replied:
Ok, maybe I wasn't clear in my comments about that test. Actually this test isn't really necessary and it looks out of scope for real content tests. We could remove it, but there's a reason I added that.

Almost all real content tests count on bundle-list --all to get a list of bundles to be tested. If by any change this command is broken and it's returning, for example, just one bundle. In this case we would run the tests only on that bundle and even if all tests are successful we can't assure that swupd is working fine for other cases.

So we don't need to run RC001 before any other test, so I didn't add it as a dependency (like we do with generate-cert). No sure if blocking other tests is the best way to go on this case.

Do you think we really need to have a strategy like you suggested.

otaviobp replied:
I realized we don't need to have this test here. We can test that every time because it's not time consuming, so we can get better results. I'll remove RC001 and improve the other tests.

castulo commented on test/real_content/swupd_update_fullfiles.bats:
```diff
@@ -0,0 +1,26 @@
+# Test massive fullfile downloads
+@test "RC004: Update from first to last with --fix" {
+	# shellcheck disable=SC2153
+	echo "Install minimal system with oldest version (${VERSION[0]})"
```
all these messages that you are printing with "echo" won't show on the test's output unless the test fails.
I would suggest to use the _print_ function provided by the test library so you can see those messages as part of the output when using _bats -t_ to run a test.

otaviobp replied:
Thanks. I'll fix that

otaviobp commented:
Fixed proposed changes and squashed last commit with the one spliting files because it makes more sense to have them together

castulo commented on test/real_content/swupd_verify_install.bats:
```diff
@@ -0,0 +1,25 @@
+	done
+
+	# shellcheck disable=SC2153
+	echo "Install complete system with newest version (${VERSION[$i]})"
```
you forgot one echo here

otaviobp replied:
Doh

jwakre commented on test/real_content/real_content_lib.bash:
```diff
@@ -0,0 +1,150 @@
+#!/bin/bash
+
+source "test/functional/testlib.bash"
```
Why did this change from: load "../functional/testlib"?

With source, I can only run the tests from the top level directory

otaviobp replied:
I just got that. load is a command in bats and this is a bash script, so we need to use source. The path thing is very inconvenient, you are right about that. Is there any way to fix it? I really don't know.

jwakre replied:
Yaaa, I see that now.... We could get the path to the script:

**SCRIPT_PATH="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"**

otaviobp commented:
I have reorganized the commits and fixed the addressed issues. I removed the test "list bundles" and added a check where this is used to make sure it's working fine.

jwakre commented on test/real_content/real_content_lib.bash:
```diff
@@ -57,7 +57,7 @@ install_bundles() {
 	fi
 
-	echo "Install bundles: $BUNDLE_LIST"
+	print "Install bundles: $BUNDLE_LIST"
```
@castulo @otaviobp I ran passing/failing tests and I do not see output from the 'print' command. Does bats actually show the output of 'print' or am I missing something here?

castulo replied:
bats will show that output on passing test when using the _-t_ option (bats -t your_test.bats), or it will show it in failing tests without using -t

jwakre replied:
ahhh I see
